modules/fargate/fargate.tf

```diff
@@ -0,0 +1,44 @@
+# EKS Fargate Pod Execution Role
+
+data "aws_iam_policy_document" "eks_fargate_pod_assume_role" {
```

Comment:
This rule is based on instructions on how to set up EKS Fargate in the [AWS documentation](https://docs.aws.amazon.com/eks/latest/userguide/fargate-getting-started.html). I don't think you can make Fargate work with EKS without it.

Reply:
Yes, you need an IAM role. But people working in very restrictive environments will have IAM roles created by a central trusted team and then developers can use them. Or permissions boundaries. See #266 and #282

Reply:
Okay, I added a variable `create_fargate_pod_execution_role` and made it `false` by default so that the role doesn't get created unless people actually want to used Fargate with their EKS cluster.